internal/controller/pkg/manager/reconciler.go

```diff
@@ -390,9 +390,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		}
 	}
 
+	// TODO(phisco): refactor these conditions to make it clearer
 	if pr.GetCondition(v1.TypeHealthy).Status == corev1.ConditionTrue {
 		p.SetConditions(v1.Healthy())
```

Comment:
Should this be moved down below the following condition?

Right now it looks like we set the provider to healthy, then emit an event if the provider is not healthy. So I think the event would never be emitted. My guess is the intention is to omit the event only once, when the provider first becomes healthy. 😄

Reply:
You are definitely right 🤦‍♂️

Reply:
Done